src/NumericalAlgorithms/Interpolation/InterpolationTargetLineSegment.hpp

```diff
@@ -0,0 +1,132 @@
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <algorithm>
+#include <pup.h>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "NumericalAlgorithms/Interpolation/SendPointsToInterpolator.hpp"
+#include "NumericalAlgorithms/Interpolation/Tags.hpp"
+#include "Parallel/ConstGlobalCache.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+namespace intrp {
+
+namespace OptionHolders {
+/// A line segment extending from `Begin` to `End`, containing `NumberOfPoints`
+/// uniformly-spaced points including the endpoints.
+struct LineSegment {
+  struct Begin {
+    using type = std::array<double, 3>;
+    static constexpr OptionString help = {"Begin"};
+  };
+  struct End {
+    using type = std::array<double, 3>;
+    static constexpr OptionString help = {"End"};
+  };
+  struct NumberOfPoints {
+    using type = size_t;
+    static constexpr OptionString help = {"NumberOfPoints"};
+    static type lower_bound() { return 2; }
+  };
+  using options = tmpl::list<Begin, End, NumberOfPoints>;
+  static constexpr OptionString help = {"Options for LineSegment"};
+
+  LineSegment(std::array<double, 3> begin_in, std::array<double, 3> end_in,
+              size_t number_of_points_in)
+      : begin(std::move(begin_in)),  // NOLINT
+        end(std::move(end_in)),      // NOLINT
+        number_of_points(number_of_points_in) {}
+  // above NOLINT for std::move of trivially copyable type.
+
+  LineSegment() = default;
+  LineSegment(const LineSegment& /*rhs*/) = default;
+  LineSegment& operator=(const LineSegment& /*rhs*/) = default;
+  LineSegment(LineSegment&& /*rhs*/) noexcept = default;
+  LineSegment& operator=(LineSegment&& /*rhs*/) noexcept = default;
+  ~LineSegment() = default;
+
+  // Need pupper so that this class can be used as a CacheTag.
+  // clang-tidy non-const reference pointer.
+  void pup(PUP::er& p) noexcept {  // NOLINT
+    p | begin;
+    p | end;
+    p | number_of_points;
+  }
+
+  std::array<double, 3> begin{};
+  std::array<double, 3> end{};
+  size_t number_of_points{};
+};
+}  // namespace OptionHolders
+
+namespace Actions {
+/// \ingroup ActionsGroup
+/// \brief Sends points on a line segment to an `Interpolator`.
+///
+/// Uses:
+/// - DataBox:
+///   - `::Tags::Domain<`VolumeDim`, Frame>`
+///   - `::Tags::Variables<typename
+///                   InterpolationTargetTag::vars_to_interpolate_to_target>`
+///
+/// DataBox changes:
+/// - Adds: nothing
+/// - Removes: nothing
+/// - Modifies:
+///   - `Tags::IndicesOfFilledInterpPoints`
+///   - `::Tags::Variables<typename
+///                   InterpolationTargetTag::vars_to_interpolate_to_target>`
+///
+/// For requirements on InterpolationTargetTag, see InterpolationTarget
+template <typename InterpolationTargetTag, typename ParallelComponentToCall,
+          size_t VolumeDim, typename Frame>
+// The only reason that there is a template parameter
```

Comment:
Is it possible to test this in the mock framework without having the ParallelComponentToCall parameter?

Reply:
If I've understood what you are trying to do correctly, what you can do is mock a component by specifying the `component_being_mocked` type alias. See, for example, `tests/Unit/IO/Observers/ObserverHelpers.hpp`. What this does is when a component calls something on a component `A`, the mocking framework actually invokes it on, say `MockA`.

Reply:
Thanks, this appears to work. Will make changes (and squash) soon...

Reply:
Great, glad to hear that works! You're welcome :)